ios_sample_test_app/ios_sample_test_app/ViewController.swift

```diff
@@ -0,0 +1,488 @@
+        view.addGestureRecognizer(viewTapGesture)
+        
+        createBranchLink()
+    }
+    
+    // ---------- Branch SDK Functions ----------
+    
+    func createBranchLink() {
+        let buo = BranchUniversalObject.init(canonicalIdentifier: "context/12345")
+        buo.title = "My Content Title"
+        buo.contentDescription = "My Content Description"
+        buo.imageUrl = "https://lorempixel.com/400/400"
+        buo.publiclyIndex = true
+        buo.locallyIndex = true
+        buo.contentMetadata.customMetadata["key1"] = "value1"
+        
+        let lp: BranchLinkProperties = BranchLinkProperties()
+        lp.channel = "facebook"
+        lp.feature = "sharing"
+        lp.campaign = "content 123 launch"
+        lp.stage = "new user"
+        lp.tags = ["one", "two", "three"]
+
+        lp.addControlParam("$desktop_url", withValue: "http://example.com/desktop")
+        lp.addControlParam("$ios_url", withValue: "http://example.com/ios")
+        lp.addControlParam("$ipad_url", withValue: "http://example.com/ios")
+        lp.addControlParam("$android_url", withValue: "http://example.com/android")
+        lp.addControlParam("$match_duration", withValue: "2000")
+
+        lp.addControlParam("custom_data", withValue: "yes")
+        
+        buo.getShortUrl(with: lp) { url, error in
+            print(url ?? "")
+            self.generatedBranchLink.text = url
+        }
+    }
+    
+    @objc func sendPurchaseEvent(_ sender: UIButton) {
+        let event = BranchEvent.customEvent(withName: "Custom Purchase Event")
+        event.currency = .USD
+        event.revenue = 5
+        event.eventDescription = "User clicked the puchase button to send a custom purchase event"
+        event.customData = [
+            "Custom_Event_Property_key1" : "Custom_Event_Property_val1",
+            "App_Type" : "Ios_Sample_App"
+        ]
+        event.logEvent()
+        // logout
+        Branch.getInstance().logout() // or .logoutWithCallback() to customize further
+        
+        self.showToast(message: "Purchase Event sent!", font: UIFont(name: "Raleway-Regular", size: 16)!)
+    }
+    
+    @objc func sendAddToCartEvent(_ sender: UIButton) {
+        let event = BranchEvent.standardEvent(.addToCart)
+        event.currency = .USD
+        event.revenue = 0
+        event.eventDescription = "User clicked add to cart button to send the standard Add to Cart Branch Event"
+        event.customData = [
+            "Custom_Event_Property_key1" : "Custom_Event_Property_val1",
+            "App_Type" : "Ios_Sample_App"
+        ]
+        event.logEvent()
+        
+        // logout
+        Branch.getInstance().logout() // or .logoutWithCallback() to customize further
+        
+        self.showToast(message: "Add to Cart event sent!", font: UIFont(name: "Raleway-Regular", size: 16)!)
+    }
+    
+    @objc func shareBranchLink() {
+        let buo = BranchUniversalObject.init(canonicalIdentifier: "context/23456")
```

Comment:
```suggestion
        let buo = BranchUniversalObject(canonicalIdentifier: "context/23456")
```
You can initialize without the init as well.

Reply:
Removed .init from BUO events